app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/DraftExciseMovementController.scala

```diff
@@ -80,6 +80,30 @@ class DraftExciseMovementController @Inject()(
         result.merge
     }
 
+
+
+  }
+
+  private def submitSaveAudit(request: ParsedXmlRequest[_], ern: String, boxId: Option[String], message: IE815Message)(implicit hc: HeaderCarrier): EitherT[Future, Result, Movement] = {
+    EitherT {
+      submissionMessageService.submit(request, ern).flatMap {
+        case Left(result) => auditService.auditMessage(message, "Failed to Submit")
+          Future.successful(Left(result))
+        case Right(_) => saveMovement(boxId, message).value.map {
+          case Left(result) => auditService.auditMessage(message, "Failed to Save")
+            Left(result)
+          case Right(movement) => auditService.auditMessage(message)
+            Right(movement)
+        }
+      }.recover {
```

Comment:
do we need this recover? Submit will cache any exception for the message submission and the NRS and return a left. saveMovement will to the same. So this recover may be redundant. auditMessage also will cache an exception and return ( HandlerResult.Failure).

Also what about if we put the audit inside the saveMovement, so this function is less indenting, less cases and more readable?

Reply:
My instinct was also it's not needed but it is covered by the "sends a failure audit on unexpected error in movement service" test.

Have moved the auditing.